src/components/Login/Login.js

```diff
@@ -1,26 +1,66 @@
-import React from 'react';
+import React, { useEffect, useRef } from 'react';
+import fetch from 'isomorphic-fetch';
 import styles from  './Login.css';
 import useStoreon from 'storeon/react'
 import Header from '../common/Header';
 
 const Login = () => {
     const { user, dispatch } = useStoreon('user');
+    const usernameEl = useRef(null);
+    const passwordEl = useRef(null);
+    
+    useEffect(()=>{
+        console.log('useEffect');
+        console.log('user', user);
+    });
+
     return(
         <div className={styles['login-page']}>
             <Header title={'View service details'} />
             <div className={styles["form"]}>
                 <div className={styles['input-row']}>
-                    <input type="text" name="username" placeholder="Username" />
+                    <input ref={usernameEl} type="text" name="username" placeholder="Username" />
                 </div>
                 <div className={styles['input-row']}>
-                    <input type="password" name="password" placeholder="Password" />
+                    <input ref={passwordEl} type="password" name="password" placeholder="Password" />
                 </div>
                 <div className={styles['input-row']}>
-                    <button type="button" onClick={ ()=> dispatch('user:loggedin') }>Login</button>
+                    <button type="button" onClick={onLogin.bind(null, dispatch, usernameEl, passwordEl)}>Login</button>
                 </div>
             </div>
         </div>
     )
 }
 
+const onLogin = (dispatch, usernameEl, passwordEl) => {
```

Comment:
Instead of passing element here, it would be good to pass the values only

Reply:
Tried. Not working. Looks like while binding el.current.value is not available. Hence continuing to pass el.